ReactAndroid/src/main/java/com/facebook/react/modules/core/Timing.java

```diff
@@ -107,13 +120,84 @@ public void doFrame(long frameTimeNanos) {
     }
   }
 
+  private class IdleFrameCallback implements Choreographer.FrameCallback {
+
+    @Override
+    public void doFrame(long frameTimeNanos) {
+      if (isPaused.get()) {
+        return;
+      }
+
+      // If the JS thread is busy for multiple frames we cancel any other pending runnable.
+      if (mCurrentIdleCallbackRunnable != null) {
+        mCurrentIdleCallbackRunnable.cancel();
+      }
+
+      mCurrentIdleCallbackRunnable = new IdleCallbackRunnable(frameTimeNanos);
+      getReactApplicationContext().runOnJSQueueThread(mCurrentIdleCallbackRunnable);
```

Comment:
Sorry I didn't catch this before: can you remind me why we're posting a runnable to the JS thread that then posts more runnables (via getJSModule(...).callIdleCallbacks)? Why not just callIdleCallbacks here?